From stable-bounces@linux.kernel.org  Sat Feb  4 02:36:34 2006
Date: Sat, 04 Feb 2006 02:29:45 -0800 (PST)
From: "David S. Miller" <davem@davemloft.net>
To: stable@kernel.org
Cc: 
Subject: [ICMP]: Fix extra dst release when ip_options_echo fails

From: Herbert Xu <herbert@gondor.apana.org.au>

When two ip_route_output_key lookups in icmp_send were combined I
forgot to change the error path for ip_options_echo to not drop the
dst reference since it now sits before the dst lookup.  To fix it we
simply jump past the ip_rt_put call.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: David S. Miller <davem@davemloft.net>

 net/ipv4/icmp.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.15.2/net/ipv4/icmp.c
===================================================================
--- linux-2.6.15.2.orig/net/ipv4/icmp.c
+++ linux-2.6.15.2/net/ipv4/icmp.c
@@ -524,7 +524,7 @@ void icmp_send(struct sk_buff *skb_in, i
 					  iph->tos;
 
 	if (ip_options_echo(&icmp_param.replyopts, skb_in))
-		goto ende;
+		goto out_unlock;
 
 
 	/*
